File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/intro/AllowMetadataUploadIntroSlide.java
package sk.styk.martin.apkanalyzer.activity.intro;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import com.github.paolorotolo.appintro.AppIntroBaseFragment;
import com.github.paolorotolo.appintro.ISlidePolicy;
import com.github.paolorotolo.appintro.model.SliderPage;

import sk.styk.martin.apkanalyzer.R;

public class AllowMetadataUploadIntroSlide extends AppIntroBaseFragment implements ISlidePolicy {
    private RadioGroup groupAllowUpload;
    private RadioButton buttonAllowUpload;
    private RadioButton buttonNotAllowUpload;

    private boolean isSelected;

    public static AllowMetadataUploadIntroSlide newInstance(SliderPage sliderPage) {
        AllowMetadataUploadIntroSlide slide = new AllowMetadataUploadIntroSlide();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, sliderPage.getTitleString());
        args.putString(ARG_TITLE_TYPEFACE, sliderPage.getTitleTypeface());
        args.putString(ARG_DESC, sliderPage.getDescriptionString());
        args.putString(ARG_DESC_TYPEFACE, sliderPage.getDescTypeface());
        args.putInt(ARG_DRAWABLE, sliderPage.getImageDrawable());
        args.putInt(ARG_BG_COLOR, sliderPage.getBgColor());
        args.putInt(ARG_TITLE_COLOR, sliderPage.getTitleColor());
        args.putInt(ARG_DESC_COLOR, sliderPage.getDescColor());
        slide.setArguments(args);

        return slide;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_intro_1;
    }


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View root = super.onCreateView(inflater, container, savedInstanceState);
        groupAllowUpload = root.findViewById(R.id.allow_upload_group);
        buttonAllowUpload = root.findViewById(R.id.allow_upload);
        buttonNotAllowUpload = root.findViewById(R.id.not_allow_upload);

        return root;
    }

    @Override
    public boolean isPolicyRespected() {
        return buttonAllowUpload.isChecked() || buttonNotAllowUpload.isChecked();
    }

    @Override
    public void onUserIllegallyRequestedNextPage() {
        // User illegally requested next slide
        Snackbar.make(getActivity().findViewById(android.R.id.content), R.string.intro_upload_require_input, Snackbar.LENGTH_LONG).show();
    }

    public boolean isUploadAllowed() {
        return buttonAllowUpload.isChecked();
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/intro/AllowMetadataUploadIntroSlide.kt
package sk.styk.martin.apkanalyzer.activity.intro

import android.os.Bundle
import android.support.design.widget.Snackbar
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RadioButton
import android.widget.RadioGroup

import com.github.paolorotolo.appintro.AppIntroBaseFragment
import com.github.paolorotolo.appintro.ISlidePolicy
import com.github.paolorotolo.appintro.model.SliderPage

import sk.styk.martin.apkanalyzer.R

/**
 * @author Martin Styk
 * @version 13.12.2017
 */
class AllowMetadataUploadIntroSlide : AppIntroBaseFragment(), ISlidePolicy {
    private var groupAllowUpload: RadioGroup? = null
    private var buttonAllowUpload: RadioButton? = null
    private var buttonNotAllowUpload: RadioButton? = null

    private val isSelected: Boolean = false

    val isUploadAllowed: Boolean
        get() = buttonAllowUpload!!.isChecked

    override fun getLayoutId(): Int {
        return R.layout.fragment_allow_metadata_upload_intro_slide
    }


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val root = super.onCreateView(inflater, container, savedInstanceState)
        groupAllowUpload = root!!.findViewById(R.id.allow_upload_group)
        buttonAllowUpload = root.findViewById(R.id.allow_upload)
        buttonNotAllowUpload = root.findViewById(R.id.not_allow_upload)

        return root
    }

    override fun isPolicyRespected(): Boolean {
        return buttonAllowUpload!!.isChecked || buttonNotAllowUpload!!.isChecked
    }

    override fun onUserIllegallyRequestedNextPage() {
        // User illegally requested next slide
        Snackbar.make(activity.findViewById(android.R.id.content), R.string.intro_upload_require_input, Snackbar.LENGTH_SHORT).show()
    }

    companion object {

        fun newInstance(sliderPage: SliderPage): AllowMetadataUploadIntroSlide {
            val slide = AllowMetadataUploadIntroSlide()
            val args = Bundle()
            args.putString(AppIntroBaseFragment.ARG_TITLE, sliderPage.titleString)
            args.putString(AppIntroBaseFragment.ARG_TITLE_TYPEFACE, sliderPage.titleTypeface)
            args.putString(AppIntroBaseFragment.ARG_DESC, sliderPage.descriptionString)
            args.putString(AppIntroBaseFragment.ARG_DESC_TYPEFACE, sliderPage.descTypeface)
            args.putInt(AppIntroBaseFragment.ARG_DRAWABLE, sliderPage.imageDrawable)
            args.putInt(AppIntroBaseFragment.ARG_BG_COLOR, sliderPage.bgColor)
            args.putInt(AppIntroBaseFragment.ARG_TITLE_COLOR, sliderPage.titleColor)
            args.putInt(AppIntroBaseFragment.ARG_DESC_COLOR, sliderPage.descColor)
            slide.arguments = args

            return slide
        }
    }
}